src/processout/processout.ts

```diff
@@ -297,20 +297,48 @@ module ProcessOut {
 
                     request.onload(<ProgressEvent>e);
                 else
-                    error(request, e);
-            };
+                    error(request, e, this.getApiRequestErrorCode(request.status));
+            }.bind(this);
             request.ontimeout  = function() {}; // Prevent IE from aborting
             request.onprogress = function() {}; // ''
             request.onabort = function() {
                 // We want to retry the call: in some cases IE fails
                 // to finish requests
-                if (retry > 3) error(request, null);
+                if (retry > 3) error(request, null, "processout-js.aborted-retries-exceeded");
                 else           this.request(method, path, data, success, error, retry + 1);
             }.bind(this);
 
             request.send(JSON.stringify(data));
         }
 
+        /**
+         * Takes an HTTP Status code and returns an ApiRequestError
+         * @param {XMLHttpRequest['status']} httpStatusCode
+         * @return {ApiRequestError}
+         */
+        protected getApiRequestErrorCode(httpStatusCode: XMLHttpRequest['status']): ApiRequestError {
+            switch (httpStatusCode) {
+                case 0:
+                    return "processout-js.network-issue"
+                case 501:
+                    return "processout-js.not-implemented"
+                case 502:
+                    return "processout-js.invalid-response"
```

Comment:
Not sure what `invalid-response` maps to, but this might also indicate something similar to 503 - depends on how our infra behaves in some scenarios.